TextToSpeech/impl/TTSSpeaker.cpp

```diff
@@ -224,16 +226,10 @@ bool TTSSpeaker::isSpeaking(uint32_t id) {
 bool TTSSpeaker::cancelCurrentSpeech(uint32_t id) {
     TTSLOG_VERBOSE("Cancelling current speech");
     bool status = false;
-    if(m_isSpeaking) {
+    if(m_isSpeaking && m_currentSpeech && (m_currentSpeech->id == id)) {
```

Comment:
This API can be changed as "cancelSpeech(uint32_t id)". If 0 is passed as argument, it should cancel any speech whichever is in progress - this is required for TTSSpeaker::reset(). If it is non-zero, it should match the current speech and cancel only if it succeeds.You can have a validation in TTSManager for (speechid != 0), to avoid any clients cancelling speech from someother client

With the present logic in the review, functionality of reset() is broken i.e Speaker couldn't do preemptive speeches.

Reply:
I have added the id==0 handling. So this will make sure reset() works.